great_tables/_helpers.py

```diff
@@ -518,6 +521,458 @@ def _get_font_stack(name: FontStackName = "system-ui", add_emoji: bool = True) -
+    concluding step for display of the units in HTML is to use the `to_html()` method.
+
+    ```{python}
+    #| echo: false
+
+    from great_tables import GT, style, loc
+    import polars as pl
+
+    units_tbl = pl.DataFrame(
+        {
+            "rule": [
+                "unit with superscript",
+                "unit with subscript",
+                "unit with subscript and superscript",
+                "unit with subscript and superscript (using overstriking)",
+                "slashed-unit shorthand for a '-1' exponent",
+                "slashes between units normalized",
+                "multiple inline units, separating by a space",
+                "use of a number allowed with previous rules",
+                "use of 'x' preceding number to form scalar multiplier",
+                "hyphen is transformed to minus sign when preceding a unit",
+                "italicization of base unit",
+                "emboldening of base unit",
+                "italicizing and emboldening of base unit",
+                "styling of subscripts and superscripts",
+                "transformation of common units from ASCII to preferred form - 1",
+                "transformation of common units from ASCII to preferred form - 2",
+                "transformation of common units from ASCII to preferred form - 3",
+                "insertion of common symbols and Greek letters via `:[symbol name]:` - 1",
+                "insertion of common symbols and Greek letters via `:[symbol name]:` - 2",
+                "insertion of common symbols and Greek letters via `:[symbol name]:` - 3",
+                "use of chemical formulas via `%[chemical formula]%`",
+            ],
+            "input": [
+                "m^2",
+                "h_0",
+                "h_0^3",
+                "h[_0^3]",
+                "/s",
+                "t_0 / t_n",
+                "kg^2 m^-1",
+                "10^3 kg^2 m^-1",
+                "x10^3 kg^2 m^-1",
+                "-h^2",
+                "*m*^2",
+                "**m**^2",
+                "_**m**_^2",
+                "h_*0*^**3**",
+                "ug",
+                "um",
+                "degC",
+                ":angstrom:_2",
+                ":permille:C",
+                ":Sigma:^2 :delta:^-1 :alpha:^-1",
+                "g/L %C6H12O6%",
+            ],
+        }
+    ).with_columns(output=pl.col("input"))
+
+    (
+        GT(units_tbl)
+        .fmt_units(columns="output")
+        .tab_style(
+            style=style.text(font="courier"),
+            locations=loc.body(columns="input")
+        )
+    )
+    ```
+    """
+
+    # Get a list of raw tokens
+    tokens_list = _generate_tokens_list(units_notation=units_notation)
+
+    # Initialize a list to store the units
+    units_list = []
+
+    if len(tokens_list) == 0:
+        return UnitDefinitionList(units_list=[])
+
+    for i in range(len(tokens_list)):
```

Comment:
Because...

* this loop creates a UnitDefinition object for each token
* UnitDefinition will likely never be instantiated directly (because you have to input token, along with everything else this loop calculates)

It might be good as a constructor on `UnitDefinition`?

e.g.

```python
@dataclass
UnitDefinition:

    ...


    @classmethod
    def from_token(cls, token: str) -> UnitDefinition:
        # logic from the loop here ----

        unit_subscript = None
        sub_super_overstrike = False
        chemical_formula = False
        exponent = None

        ...
        
        return cls(token, ...)
```

Reply:
Now implemented.